src/main/java/net/dv8tion/jda/internal/entities/GuildImpl.java

```diff
@@ -245,71 +190,22 @@ public void invalidate()
     public void uncacheChannel(GuildChannel channel, boolean keepThreads)
     {
         long id = channel.getIdLong();
-        switch (channel.getType())
-        {
-        case TEXT:
-            api.getTextChannelsView().remove(id);
-            this.getTextChannelsView().remove(id);
-            break;
-        case NEWS:
-            api.getNewsChannelView().remove(id);
-            this.getNewsChannelView().remove(id);
-            break;
-        case MEDIA:
-            api.getMediaChannelsView().remove(id);
-            this.getMediaChannelsView().remove(id);
-            break;
-        case FORUM:
-            api.getForumChannelsView().remove(id);
-            this.getForumChannelsView().remove(id);
-            break;
-        case VOICE:
-            api.getVoiceChannelsView().remove(id);
-            this.getVoiceChannelsView().remove(id);
-            break;
-        case STAGE:
-            api.getStageChannelView().remove(id);
-            this.getStageChannelsView().remove(id);
-            break;
-        case CATEGORY:
-            api.getCategoriesView().remove(id);
-            this.getCategoriesView().remove(id);
-            break;
-        case GUILD_NEWS_THREAD:
-        case GUILD_PUBLIC_THREAD:
-        case GUILD_PRIVATE_THREAD:
-            api.getThreadChannelsView().remove(id);
-            this.getThreadChannelsView().remove(id);
-            break;
-        }
+        if (channelCache.remove(channel.getType(), id) != null)
+            api.getChannelsView().remove(channel.getType(), id);
```

Comment:
I don't want an outdated guild instance to somehow affect the new cache tied to that channel. My worry here was that you would call `guild.uncacheChannel` twice and it might break the new cache from a new guild instance.

Reply:
Ah. Let's document that then